src/stan/variational/advi.hpp

```diff
@@ -334,6 +357,14 @@ namespace stan {
 
         // draw more samples from posterior and write on subsequent lines
         if (out_stream_) {
+          if (print_stream_) {
```

Comment:
`advi.hpp` makes no assumptions on any dependencies among `print_stream_`, `out_stream_`, and `diagnostic_stream_`.

The code doesn't do `if (cond 1 && cond2)` because it checks `cond2` multiple times within the control flow of the `if (cond1)` loop. Reimplementing it would change it to be `if (cond1 && cond 2) print stuff ... if (cond1) draw stuff ... if (cond1 && cond2) print stuff`. I don't know if that's more readable.